### DISCUSSION TITLE
Vsync/frame rates not working correct

### DISCUSSION THREAD

**Discussion by Sprink02** [ppy/osu#24883] [Q&A]
### Type

Game behaviour

### Bug description

Osu seems to not like the frame limit setting. The 2 settings that conflict is Nvidia's vsync driver setting and the ingame setting

Ex: (In game setting / Driver setting) > outcome
(Vsync on / Use 3d app setting) > Vsync on 165 fps
(Vsync on / Vsync fast/off) > Vsync off 1000 fps
(2x frame limit / Use 3d app setting) > Vsync on 165 fps
(2x frame limit/ Vsync fast/off) > Vsync fast/off 330 fps

Game settings:
Multithreaded
Dx 11 render
Borderless Windowed

Pc specs:
Laptop
Ryzen 7 4800H
Rtx 2060 mobile
no mux switch
Running on external monitor at 165 hz
Internal monitor at 120 hz
Running game on Rtx 2060

### Version

2023.914.0-lazer

### Logs

[performance.log](https://github.com/ppy/osu/files/12694420/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12694421/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12694422/updater.log)
[database.log](https://github.com/ppy/osu/files/12694423/database.log)
[input.log](https://github.com/ppy/osu/files/12694424/input.log)
[network.log](https://github.com/ppy/osu/files/12694425/network.log)

**Comment by peppy** [comment]
Looks roughly correct?

You're probably looking at the update thread, not draw.

**Reply by Sprink02** [reply]
![image](https://github.com/ppy/osu/assets/69661817/75430512-557f-4aa5-b358-21c52f89fc55)
![image](https://github.com/ppy/osu/assets/69661817/3e30c02e-575d-4f7c-98de-c0280ae0877e)
![image](https://github.com/ppy/osu/assets/69661817/64432be1-49d6-4e57-8714-fa4e29df518d)
![image](https://github.com/ppy/osu/assets/69661817/1d73f755-4f28-42d7-847d-8f36b8c38958)

4 pics confirming the details above. :)
Currently attempting different fixes. Currently making a doc :)

**Reply by Sprink02** [reply]
[Osu Tests.xlsx](https://github.com/ppy/osu/files/12705645/Osu.Tests.xlsx)

Tests that have all the data likely needed from my pc.

**Reply by peppy** [reply]
This all looks correct.

Maybe you can explain what you're hoping for here.

**Reply by longnguyen2004** [reply]
Looks correct to me, maybe aside from the last pic, where it looks like VSync is on even though the limiter is set to 2xVSync. Fairly certain not lazer's fault, since I don't have that issue.

**Reply by peppy** [reply]
> Looks correct to me, maybe aside from the last pic, where it looks like VSync is on even though the limiter is set to 2xVSync. Fairly certain not lazer's fault, since I don't have that issue.

It's quite possible even that is correct if the game isn't hitting the "2x" limit, and vsyncing back to 1x (correct behaviour in such cases).

**Reply by Sprink02** [reply]
The main problem Im having is when trying a higher frame rate with Nvidia being auto it is not achieving that frame rate. Its just something that I believe should be handled by the game and not having to go into the Nvidia control panel to fix it.

**Reply by longnguyen2004** [reply]
Drivers can do all sorts of wacky stuff under the hood and there's nothing that the game can do, because it's invisible to the game itself. Working around driver quirks isn't something the team has to do, unless it's some critical game breaking bugs.
